controllers/storagecluster/external_resources.go

```diff
@@ -561,9 +561,13 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 		r.Log.Error(err, "Failed to create needed StorageClasses.")
 		return err
 	}
-	if err = r.setRookCSICephFS(enableRookCSICephFS, instance); err != nil {
-		r.Log.Error(err, "Failed to set RookEnableCephFSCSIKey to EnableRookCSICephFS.", "RookEnableCephFSCSIKey", rookEnableCephFSCSIKey, "EnableRookCSICephFS", enableRookCSICephFS)
-		return err
+	// We do not want to disable CephFS csi driver in consumer mode since
+	// CephFS storageclass is available by default.
+	if !IsOCSConsumerMode(instance) {
+		if err = r.setRookCSICephFS(enableRookCSICephFS, instance); err != nil {
+			r.Log.Error(err, "Failed to set RookEnableCephFSCSIKey to EnableRookCSICephFS.", "RookEnableCephFSCSIKey", rookEnableCephFSCSIKey, "EnableRookCSICephFS", enableRookCSICephFS)
+			return err
+		}
```

Comment:
What if we just enable the flag when we are in `ConsumerMode`? If you see the function, `setRookCSICephFS()`, it won't do any updation if the flag is already true, thus the code look much cleaner.

```suggestion
	if IsOCSConsumerMode(instance) {
	       enableRookCSICephFS = true
	}
```